src/Lecture3.hs

```diff
@@ -210,8 +240,24 @@ types that can have such an instance.
 -- instance Foldable Weekday where
 -- instance Foldable Gold where
 -- instance Foldable Reward where
--- instance Foldable List1 where
--- instance Foldable Treasure where
+
+instance Foldable List1 where
+  foldr :: (a -> b -> b) -> b -> List1 a -> b
+  foldr reducer acc (List1 x [])     = reducer x acc
+  foldr reducer acc (List1 x (y:ys)) = reducer x (foldr reducer acc (List1 y ys))
```

Comment:
You don't need to implement a recursive implementation here manually. You can use the `Foldable` instance for lists